CondTools/SiPhase2Tracker/plugins/SiPhase2BadStripChannelBuilder.cc

```diff
@@ -0,0 +1,295 @@
+  enum badChannelAlgo { NAIVE = 1, RANDOM = 2, NONE = 99 };
+}
+
+class SiPhase2BadStripChannelBuilder : public ConditionDBWriter<SiStripBadStrip> {
+public:
+  explicit SiPhase2BadStripChannelBuilder(const edm::ParameterSet&);
+  ~SiPhase2BadStripChannelBuilder() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  std::unique_ptr<SiStripBadStrip> getNewObject() override;
+
+  void algoBeginRun(const edm::Run& run, const edm::EventSetup& es) override {
+    if (!tTopo_) {
+      tTopo_ = std::make_unique<TrackerTopology>(es.getData(topoToken_));
+      tGeom_ = std::make_unique<TrackerGeometry>(es.getData(geomToken_));
+    }
+  };
+
+  void algoAnalyze(const edm::Event& event, const edm::EventSetup& es) override {
+    edm::Service<edm::RandomNumberGenerator> rng;
+    if (!engine_) {
+      engine_ = &rng->getEngine(event.streamID());
+    }
+  }
+
+  std::map<unsigned short, unsigned short> clusterizeBadChannels(
+      const std::vector<Phase2TrackerDigi::PackedDigiType>& maskedChannels);
+
+  // ----------member data ---------------------------
+  std::unique_ptr<TrackerTopology> tTopo_;
+  std::unique_ptr<TrackerGeometry> tGeom_;
+  CLHEP::HepRandomEngine* engine_;
+
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  const bool printdebug_;
+  const unsigned int popConAlgo_;
+  const float badComponentsFraction_;
+  badChannelAlgo theBCAlgo_;
+};
+
+//__________________________________________________________________________________________________
+SiPhase2BadStripChannelBuilder::SiPhase2BadStripChannelBuilder(const edm::ParameterSet& iConfig)
+    : ConditionDBWriter<SiStripBadStrip>(iConfig),
+      topoToken_(esConsumes<edm::Transition::BeginRun>()),
+      geomToken_(esConsumes<edm::Transition::BeginRun>()),
+      printdebug_(iConfig.getUntrackedParameter<bool>("printDebug", false)),
+      popConAlgo_(iConfig.getParameter<unsigned int>("popConAlgo")),
+      badComponentsFraction_(iConfig.getParameter<double>("badComponentsFraction")) {
+  if (badComponentsFraction_ > 1. || badComponentsFraction_ < 0.) {
+    throw cms::Exception("Inconsistent configuration")
+        << "[SiPhase2BadStripChannelBuilder::c'tor] the requested fraction of bad components is unphysical. \n";
+  }
+  theBCAlgo_ = static_cast<badChannelAlgo>(popConAlgo_);
+}
+
+//__________________________________________________________________________________________________
+std::unique_ptr<SiStripBadStrip> SiPhase2BadStripChannelBuilder::getNewObject() {
+  using Phase2TrackerGeomDetUnit = PixelGeomDetUnit;
+  edm::LogInfo("SiPhase2BadStripChannelBuilder") << "... creating dummy SiStripBadStrip Data" << std::endl;
+
+  auto obj = std::make_unique<SiStripBadStrip>();
+
+  // early return with nullptr if fraction is ==0.
+  if (badComponentsFraction_ > 0.) {
+    ;
+  } else {
+    return obj;
+  }
```

Comment:
```suggestion
  if (badComponentsFraction_ == 0.f) {
    return obj;
  }
```

Reply:
thanks, will fix in the next iteration.

Reply:
done at https://github.com/mmusich/cmssw/commit/1c7926639d95113e92ba39bb20ee9d0d1aea2347